src/BotService/Infrastructure/Core/CallHandler.cs

```diff
@@ -501,18 +583,18 @@ private IParticipant GetParticipantFromMSI(uint msi)
         private MediaExtractionSettings GetMediaStreamSettings(uint msi, StartStreamExtractionBody streamBody)
         {
             ProtocolSettings protocolSettings;
-            switch (streamBody.Protocol)
+
+            if (_currentAssignedPorts.Count == _numberOfMultiviewSockets)
             {
-                case Protocol.SRT:
-                    if (!_availablePorts.TryTake(out int port))
-                    {
-                        _logger.LogWarning("[Call Handler] There are no ports available");
+                _logger.LogWarning("[Call Handler] Maximum number of extractions reached");
```

Comment:
**nitpick(blocking)**: Here we have an issue. Before getting the MediaStreamSeetings, we are creating a media extractor and adding it to the currentMediaExtractors collection. If there ain't no ports available, we will throw the exception, but we don't remove the media extractor. We must get the MediaStream Settings before adding the extractor to the currentMediaExtractors collection.